service/create/service.go

```diff
@@ -462,6 +466,70 @@ func (s *Service) Boot() {
 						return
 					}
 
+					// Create Ingress Load Balancer.
+					lbInput = LoadBalancerInput{
+						Name:        cluster.Spec.Cluster.Kubernetes.IngressController.Domain,
+						Clients:     clients,
+						Cluster:     cluster,
+						InstanceIDs: workerIDs,
+						PortsToOpen: []awsresources.PortPair{
+							awsresources.PortPair{
+								PortELB:      cluster.Spec.Cluster.Kubernetes.IngressController.SecurePort,
+								PortInstance: httpsPort,
+							},
+							awsresources.PortPair{
+								PortELB:      cluster.Spec.Cluster.Kubernetes.IngressController.InsecurePort,
+								PortInstance: httpPort,
+							},
+						},
+						SecurityGroupID: workersSecurityGroup.ID(),
+						SubnetID:        publicSubnet.ID(),
+					}
+
+					ingressLB, err := s.createLoadBalancer(lbInput)
+					if err != nil {
+						s.logger.Log("error", errgo.Details(err))
+						return
+					}
+
+					s.logger.Log("info", fmt.Sprintf("created ingress load balancer"))
+
+					// Create Record Sets.
+					recordSetInputs := []recordSetInput{
+						recordSetInput{
+							Cluster:      cluster,
+							Client:       clients.Route53,
+							Resource:     mastersLB,
+							Domain:       cluster.Spec.Cluster.Kubernetes.API.Domain,
+							HostedZoneID: hz.ID(),
+						},
+						recordSetInput{
+							Cluster:      cluster,
+							Client:       clients.Route53,
+							Resource:     mastersLB,
+							Domain:       cluster.Spec.Cluster.Etcd.Domain,
+							HostedZoneID: hz.ID(),
+						},
+						recordSetInput{
+							Cluster:      cluster,
+							Client:       clients.Route53,
+							Resource:     ingressLB,
+							Domain:       cluster.Spec.Cluster.Kubernetes.IngressController.Domain,
```

Comment:
I think there is a problem with hosted zones :( The ingress domain is within aws.gigantic.io not aws.giantswarm.io so we can we use the same HZ?